SW_VegProd.c

```diff
@@ -841,3 +865,541 @@ void get_critical_rank(void){
+ @param[in] SumGrassesFraction Value holding sum of grass if user would like it to be fixed
+ @param[in] C4Variables Array of size three holding C4 variables after being averaged by `averageClimateAcrossYears()`.
+ The elements are: 0) July precipitation, 1) mean temperature of dry quarter, 2) mean minimum temperature of February
+ @param[in] fillEmptyWithBareGround Bool value specifying whether or not to fill gaps in values with bare ground
+ @param[in] inNorthHem Bool value specifying if the current site is in the northern hemisphere
+ @param[in] warnExtrapolation Bool value specifying whether or not to warn the user when extrapolation happens
+ @param[out] grassOutput Array of size three holding estimated grass values. The elements are: 0) C3, 1) C4, 2) annual grasses
+ @param[out] RelAbundanceL0 Array of size eight holding all estimated values. The elements are:
+ 0) Succulents, 1) Forbs, 2) C3, 3) C4, 4) annual grasses, 5) Shrubs, 6) Trees, 7) Bare ground
+ @param[out] RelAbundanceL1 Array of size five holding all estimated values aside from grasses (not including sum of grasses).
+ The elements are: 0) trees, 1) shrubs 2) sum of forbs and succulents 3) overall sum of grasses 4) bare ground
+
+ @note This function uses equations developed by
+ Paruelo & Lauenroth (1996) @cite paruelo1996EA and,
+ for C4 grasses, an equation by Teeri & Stowe (1976) @cite teeri1976O.
+ */
+
+void estimatePotNatVegComposition(double meanTemp_C, double PPT_cm, double meanTempMon_C[],
+    double PPTMon_cm[], double inputValues[], double shrubLimit, double SumGrassesFraction,
+    double C4Variables[], Bool fillEmptyWithBareGround, Bool inNorthHem, Bool warnExtrapolation,
+    double *grassOutput, double *RelAbundanceL0, double *RelAbundanceL1) {
+
+    const int nTypes = 8;
+    int winterMonths[3], summerMonths[3];
+
+    // Indices both single value and arrays
+    int index, succIndex = 0, forbIndex = 1, C3Index = 2, C4Index = 3, grassAnn = 4,
+    shrubIndex = 5, treeIndex = 6, bareGround = 7, grassEstimSize = 0, overallEstimSize = 0,
+    julyMin = 0, frostFreeDays = 1, degreeAbove65 = 2, estimIndicesNotNA = 0, grassesEstim[3],
+    overallEstim[nTypes], iFixed[nTypes], iFixedSize = 0,
+    isetIndices[3] = {grassAnn, treeIndex, bareGround};
+
+    // Totals of different areas of variables
+    double totalSumGrasses = 0., inputSumGrasses = 0., tempDiffJanJul,
+    summerMAP = 0., winterMAP = 0., C4Species = SW_MISSING, C3Grassland, C3Shrubland,
+    estimGrassSum = 0, finalVegSum = 0., estimCoverSum = 0., tempSumGrasses = 0.,
+    estimCover[nTypes], initialVegSum = 0., tempSwapValue, fixedValuesSum = 0;
+
+    Bool fixSumGrasses = (Bool) (!missing(SumGrassesFraction)),
+    isGrassIndex = swFALSE, tempShrubBool;
+
+    // Loop through inputValues and get the total
+    for(index = 0; index < nTypes; index++) {
+        if(!missing(inputValues[index])) {
+            initialVegSum += inputValues[index];
+        }
+    }
+
+    if(initialVegSum > 1.) {

```

Comment:
- [ ] Please replace `initialVegSum > 1.` with `GT(initialVegSum, 1.)` to account for floating point issues
- [ ] Please create a death test that is triggered by this situation